cdap-api/src/main/java/io/cdap/cdap/api/plugin/PluginClass.java

```diff
@@ -43,17 +44,19 @@
     this.type = Plugin.DEFAULT_TYPE;
     this.name = null;
     this.description = "";
+    this.category = null;
     this.className = null;
     this.configFieldName = null;
     this.properties = Collections.emptyMap();
     this.requirements = Requirements.EMPTY;
   }
 
-  public PluginClass(String type, String name, String description, String className,
+  public PluginClass(String type, String name, @Nullable String category, String className,
```

Comment:
this is technically a backwards incompatible change to an API class, as anything that was calling the previous constructor will now fail to compile, and this is in cdap-api.

It would be better to add a Builder class and deprecate these constructors.